pkg/controller/backup_deletion_controller.go

```diff
@@ -387,202 +344,160 @@ func (c *backupDeletionController) processRequest(req *velerov1api.DeleteBackupR
 			errs = append(errs, err)
 		}
 	}
-
 	return errs
 }
 
-const deleteBackupRequestMaxAge = 24 * time.Hour
-
-func (c *backupDeletionController) deleteExpiredRequests() {
-	c.logger.Info("Checking for expired DeleteBackupRequests")
-	defer c.logger.Info("Done checking for expired DeleteBackupRequests")
-
-	// Our shared informer factory filters on a single namespace, so asking for all is ok here.
-	requests, err := c.deleteBackupRequestLister.List(labels.Everything())
-	if err != nil {
-		c.logger.WithError(err).Error("unable to check for expired DeleteBackupRequests")
-		return
-	}
-
-	now := c.clock.Now()
-
-	for _, req := range requests {
-		if req.Status.Phase != velerov1api.DeleteBackupRequestPhaseProcessed {
-			continue
-		}
-
-		age := now.Sub(req.CreationTimestamp.Time)
-		if age >= deleteBackupRequestMaxAge {
-			reqLog := c.logger.WithFields(logrus.Fields{"namespace": req.Namespace, "name": req.Name})
-			reqLog.Info("Deleting expired DeleteBackupRequest")
-
-			err = c.deleteBackupRequestClient.DeleteBackupRequests(req.Namespace).Delete(context.TODO(), req.Name, metav1.DeleteOptions{})
-			if err != nil {
-				reqLog.WithError(err).Error("Error deleting DeleteBackupRequest")
-			}
-		}
-	}
-}
-
-func (c *backupDeletionController) patchDeleteBackupRequest(req *velerov1api.DeleteBackupRequest, mutate func(*velerov1api.DeleteBackupRequest)) (*velerov1api.DeleteBackupRequest, error) {
-	// Record original json
-	oldData, err := json.Marshal(req)
+func (r *backupDeletionReconciler) patchDeleteBackupRequest(ctx context.Context, req *velerov1api.DeleteBackupRequest, mutate func(*velerov1api.DeleteBackupRequest)) (*velerov1api.DeleteBackupRequest, error) {
+	patchHelper, err := patch.NewHelper(req, r.Client)
 	if err != nil {
-		return nil, errors.Wrap(err, "error marshalling original DeleteBackupRequest")
+		return nil, errors.Wrap(err, "unable to get the patch helper")
 	}
-
 	// Mutate
 	mutate(req)
-
-	// Record new json
-	newData, err := json.Marshal(req)
-	if err != nil {
-		return nil, errors.Wrap(err, "error marshalling updated DeleteBackupRequest")
-	}
-
-	patchBytes, err := jsonpatch.CreateMergePatch(oldData, newData)
-	if err != nil {
-		return nil, errors.Wrap(err, "error creating json merge patch for DeleteBackupRequest")
-	}
-
-	req, err = c.deleteBackupRequestClient.DeleteBackupRequests(req.Namespace).Patch(context.TODO(), req.Name, types.MergePatchType, patchBytes, metav1.PatchOptions{})
-	if err != nil {
-		return nil, errors.Wrap(err, "error patching DeleteBackupRequest")
+	if err := patchHelper.Patch(ctx, req); err != nil {
+		return nil, errors.Wrap(err, "error patching the deletebackuprquest")
 	}
-
 	return req, nil
 }
 
-func (c *backupDeletionController) patchBackup(backup *velerov1api.Backup, mutate func(*velerov1api.Backup)) (*velerov1api.Backup, error) {
+func (r *backupDeletionReconciler) patchBackup(ctx context.Context, backup *velerov1api.Backup, mutate func(*velerov1api.Backup)) (*velerov1api.Backup, error) {
```

Comment:
Use cluster-api's NewHelper function can simplify patch code, for example, https://github.com/vmware-tanzu/velero/blob/main/pkg/controller/download_request_controller.go#L72

Reply:
it can't be used b/c the patchHelper assumes the status is a subresource and it's not true for backup CR.